### PR TITLE
Feature/api ip4log

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Adds ip4logs to API

# Impacts
Exposes ip4log, ip4log_history and ip4log_archive to API

# Delete branch after merge
YES 

# NEWS file entries
## New Features
* Add ip4logs to API

### CODE REVIEW

File: lib/pf/UnifiedApi/Controller/Ip4logs.pm [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+=cut
+
+use strict;
+use warnings;
+use Mojo::Base 'pf::UnifiedApi::Controller::Crud';
+use pf::ip4log;
+
+has dal => 'pf::dal::ip4log';
+has id_key => 'ip4log_id';
```
Comment (rouzier) [inline lib/pf/UnifiedApi/Controller/Ip4logs.pm]: url_param_name

Comment (jrouzierinverse) [inline lib/pf/UnifiedApi/Controller/Ip4logs.pm]:
```perl
has url_param_name => 'ip4log_id';
```

File: lib/pf/UnifiedApi/Controller/Ip4logs.pm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+has resource_id => 'tenant_id';
```
Comment (jrouzierinverse) [inline lib/pf/UnifiedApi/Controller/Ip4logs.pm]:
```perl
has primary_key => 'ip';
```

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by extrafu** [comment]
@jrouzierinverse Please review and merge.

File: lib/pf/UnifiedApi.pm [hunk 3]
```diff
@@ -112,6 +112,19 @@ our @API_V1_ROUTES = (
...
     },
+    
+    
+    {
+        controller  => 'Ip4logs',
+        collection => {
+            subroutes    => {
+                'history/:search' => { get => 'history' },
+                'archive/:search' => { get => 'archive' },
+                'search/:search' => { get => 'search' },
```
Comment (jrouzierinverse) [inline lib/pf/UnifiedApi.pm]: Rename search route and method

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by extrafu** [comment]
@jrouzierinverse Please merge if all is good. We need this done.

File: lib/pf/UnifiedApi/Controller/Ip4logs.pm [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+has dal => 'pf::dal::ip4log';
+has url_param_name => 'ip4log_id';
+has primary_key => 'ip';
+
+sub open {
+    my ($self) = @_;
+    my $search = $self->param('search');
+    my @iplog = pf::ip4log::list_open($search);
+    return $self->render(json => $iplog[0]) if $iplog[0];
```
Comment (rouzier) [inline lib/pf/UnifiedApi/Controller/Ip4logs.pm]: Put this as an {item => }
```perl
return $self->render(json => {item => $iplog[0]}) if $iplog[0];
 ```

**Review by rouzier — CHANGES_REQUESTED** [review]

**Pull request merged**